src/django_otp/admin.py

```diff
@@ -16,6 +18,45 @@ def _admin_template_for_django_version():
     return 'otp/admin111/login.html'
 
 
+def user_model_search_fields(field_names):
+    """
+    Check whether the provided field names exist, and return a tuple of
+    search fields and help text for the validated field names.
+
+    Parameters:
+    field_names (list of str): A list of field names to search for in the user model.
+
+    Returns:
+    tuple: A tuple containing:
+        - search_fields (list of str): A list of search fields formatted for querying.
+        - search_help_text (str): A help text string describing the valid search fields.
+    """
+    User = get_user_model()
+    valid_fields = []
+    search_fields = []
+    help_texts = []
+
+    for name in field_names:
+        try:
+            field = User._meta.get_field(name)
+        except FieldDoesNotExist:
+            continue
+        else:
+            valid_fields.append(field)
+            search_fields.append("user__" + field.name)
+            help_texts.append(str(field.verbose_name))
+
+    # Join terms with commas, except use "or" before the last term
+
+    search_help_text = (
```

Comment:
Almost clever: if `help_texts` is, e.g., `['username']`, this will produce `Search by  or username`.

Reply:
Indeed I missed that. Thanks.